### PR TITLE
feat!: combine flavours, improve syntax highlights, customizability

### PR DESCRIPTION
**Pull request by nyxkrage**
Rewrite to avoid dependencies for both regular Emacs and Doom Emacs, contain all flavors in the same theme, make it closer to the other editor themes, and make it more customizable (allow user to overwrite any colors).

Since this is such a big rewrite, we should create a tag of before this gets merged.

### CODE REVIEW

**Review by backwardspy — APPROVED** [review]

**Review by sgoudham — APPROVED** [review]
mujikcboro-seriymujik

File: README.md [hunk 1]
```diff
@@ -1,69 +1,86 @@
...
+The theme can also be customzied further by changing individual colors
+``` lisp
+(catppuccin-set-color 'base "#000000") ;; change base to #000000 for the currently active flavor
+(catppuccin-set-color 'crust "#222222" 'frappe) ;; change crust to #222222 for frappe
+```
+
+## 💝 Thanks to
+
+- [Nyx](https://github.com/nyxkrage)
+- [Dracula](https://draculatheme.com/emacs)
```
Comment (nekowinston) [inline README.md:77]: Can we replace this with the github repo link?

File: catppuccin-theme.el [hunk 2]
```diff
@@ -30,846 +32,934 @@ The theme has to be reloaded after changing anything in this group."
...
+               (term-color-yellow :foreground ,ctp-peach :background ,ctp-peach)
+               (term-color-bright-yellow :foreground ,ctp-yellow :background ,ctp-yellow)
+               (term-color-blue :foreground ,ctp-blue :background ,ctp-blue)
+               (term-color-bright-blue :foreground ,ctp-sapphire :background ,ctp-sapphire)
+               (term-color-magenta :foreground ,ctp-mauve :background ,ctp-mauve)
+               (term-color-bright-magenta :foreground ,ctp-lavender :background ,ctp-lavender)
+               (term-color-cyan :foreground ,ctp-sky :background ,ctp-sky)
+               (term-color-bright-cyan :foreground ,ctp-teal :background ,ctp-teal)
+               (term-color-white :foreground ,ctp-text :background ,ctp-text)
+               (term-color-bright-white :foreground ,ctp-overlay2 :background ,ctp-overlay2)
```
Comment (nekowinston) [inline catppuccin-theme.el:847]: This should probably follow [the terminal style guide](https://github.com/catppuccin/catppuccin/blob/main/docs/style-guide.md#terminals), handling Latte differently.

**Review by nekowinston — COMMENTED** [review]
Thank you so much for taking this on, just 2 minor comments.

File: README.md [hunk 3]
```diff
@@ -1,69 +1,86 @@
...
+
+## Doom Emacs
+1. Download `catppuccin-theme.el` into `.doom.d/themes`
+1. Add the following to your `config.el`
+
+``` lisp
+(setq doom-theme 'catppuccin)
+```
+
+# Configuration
```
Comment (nyxkrage) [inline README.md:59]: Doom Emacs does not load in the functions of the theme after just using `(setq doom-theme 'catppuccin)`
So it should be mentioned that Doom Emacs users need to have `(load-theme 'catppuccin t t)` before they call any of the catppuccin functions

File: catppuccin-theme.el [hunk 4]
```diff
@@ -30,846 +32,934 @@ The theme has to be reloaded after changing anything in this group."
...
+            (pcase-dolist (`(,face . ,spec) faces)
+              (push `(,face
+                      ((((min-colors 16777216)) ; fully graphical envs
+                        ,(funcall expand-with-func 'cadr spec))
+                       (t                       ; terminal with 256 colors
+                        ,(funcall expand-with-func 'caddr spec))))
+                    whole-theme))
+            whole-theme))
 
   (apply #'custom-theme-set-variables
```
Comment (nyxkrage) [inline catppuccin-theme.el:935]: There should be added something here probably about `rustic-ansi-faces` to make the rustic terminal use catppuccin colors

**Comment by NamesCode** [comment]
Should I merge this and raise issues for these messages?

**Comment by sgoudham** [comment]
> Should I merge this and raise issues for these messages?

I believe @nyxkrage will fix based on the feedback given, no need to rush :D

**Comment by NamesCode** [comment]
Tested the theme myself and it looks great ❤️

**Comment by averageflow** [comment]
@nyxkrage , thanks for your amazing work on it! Really happy to see the development continues on this version of the lovely theme for Emacs. Since this is a major rewrite, could I suggest that you consider the issue I recently opened about Solaire mode ? https://github.com/catppuccin/emacs/issues/41 
Thanks for your work everyone

**Comment by nyxkrage** [comment]
Added some basic support for `solaire-mode` which should close #41 and should now be good to merge.

**Comment by nyxkrage** [comment]
This PR would also close #22 and #18

**Review by nekowinston — APPROVED** [review]

**Comment by NamesCode** [comment]
> Did the MELPA release get tagged as 1.0.0? If so, we should probably bump this to 2.0.0 and tag the previous commit.

No, it didn't. I can now though however. At the time I remember saying something along the lines of "who will actually miss it" which was very naive of me

**Pull request merged**